docs/en/observability/monitor-java-app.asciidoc

```diff
@@ -0,0 +1,2005 @@
+profiling of inferred spans] by setting the following.
+
+[source,bash]
+----
+ELASTIC_APM_PROFILING_INFERRED_SPANS_ENABLED=true
+----
+
+This mechanism uses the
+https://github.com/jvm-profiling-tools/async-profiler[async profiler] to
+create spans without you having to instrument anything allowing you to
+find bottlenecks faster.
+
+[discrete]
+==== Log correlation
+
+By configuring the agent, let's ease the
+https://www.elastic.co/guide/en/apm/agent/java/current/config-logging.html#config-enable-log-correlation[correlation
+of logs] by adding the transaction ids to our logs.
+
+[source,bash]
+----
+ELASTIC_APM_ENABLE_LOG_CORRELATION=true
+----
+
+You can now check the generated log files that are sent
+to {es} via Filebeat. An entry looks like this.
+
+[source,json]
+----
+{
+  "@timestamp": "2020-07-13T12:03:22.491Z",
+  "log.level": "INFO",
+  "message": "GET / 200 0:0:0:0:0:0:0:1 \"curl/7.64.1\" 0",
+  "service.name": "my-javalin-app",
+  "event.dataset": "my-javalin-app.log",
+  "process.thread.name": "qtp34871826-36",
+  "log.logger": "de.spinscale.javalin.Log4j2RequestLogger",
+  "trace.id": "ed735860ec0cd3ee3bdf80ed7ea47afb",
+  "transaction.id": "8af7dff698937dc5"
+}
+----
+
+Having the `trace.id` and `transaction.id` added, in the case of an error
+you will get an `error.id` field.
+
+[IMPORTANT]
+=====
+We have not covered the
+https://www.elastic.co/guide/en/apm/agent/java/current/opentracing-bridge.html[Elastic
+APM OpenTracing bridge] or looked into the
+https://www.elastic.co/guide/en/apm/agent/java/current/metrics.html[additional
+metrics] the agent provides, which allows us to take a look at things
+like garbage collection or the memory footprint of our application.
+=====
+
+[discrete]
+=== Step 7: Ingest Uptime data
+
+There are some basic monitoring capabilities in our application so far. We index
+logs (with traces), we index metrics, and we even can look in our app to
+figure out single performance bottlenecks thanks to APM. However, there is
+still one weak spot. Everything done so far was within the application, and all
+the users are reaching the application from the internet.
+
+How about checking if our users have the same experience that our APM data
+is suggesting to us. Imagine having a lagging load balancer fronting your app,
+that costs you an additional 50ms per request. That would be devastating. Or
+TLS negotiation being costly. Even though none of those external events is your fault, you will
+still be impacted by this and should try to mitigate those. This means you
+need to know about them first.
+
+https://www.elastic.co/uptime-monitoring[Uptime] not only enables you to monitor the availability
+of a service, but also graph latencies over time, and get notified about expiring TLS certificates.
+
+[discrete]
+==== Setup
+
+Let’s first download {heartbeat} (the polling component) and configure it
+to check for our application. For details, see <<install-heartbeat,Install {heartbeat}>>.
```

Comment:
Same comment as earlier. I would prefer that this document be self-contained. We can add related links for more info, but shouldn't send users out to a separate task flow to complete these steps.